geocodeTree.py

```diff
@@ -10,17 +10,31 @@ class GeocodeTree:
     def __init__(self):
         self.G = cf.graph()
         self.tree_count = 0
+        self.tree_list = []
 
-    def add_tree(self, coord):
+    def add_tree(self, coord, start_node, end_node):
+        self.start_node = start_node
+        self.end_node = end_node
         self.tree_count += 1
         tree_name = f"t-{self.tree_count}"
-        nearest_node = ox.nearest_nodes(self.G, coord[0], coord[1])
+        self.tree_list.append(tree_name)
         self.G.add_node(tree_name, x=coord[1], y=coord[0], street_count=3)
-        self.G.add_edge(tree_name, nearest_node)
+        # self.G.add_edge(tree_name, nearest_node)
+
         print(self.G.nodes)
         return self.G
 
     def save_and_return(self):
+        for i in range(len(self.tree_list)-1):
+            self.G.add_edge(self.tree_list[i], self.tree_list[i+1])
+            self.G.add_edge(self.tree_list[i+1], self.tree_list[i])
+
+        self.G.add_edge(self.tree_list[0], self.start_node)
+        self.G.add_edge(self.start_node, self.tree_list[0])
+        self.G.add_edge(self.tree_list[-1], self.end_node)
+        self.G.add_edge(self.end_node, self.tree_list[-1])
```

Comment:
In this code snippet,

A for loop is run, to connect all consecutive tree nodes with each other bidirectionally.

Before : T1, T2, T3    (not connected)
After : T1 <=> T2 <=> T3  (connected bidirectionally)

Then the  first tree_node is connected to the start_node and the last tree_node is connected to the last_node of the stretch.

first_node <=> T1 <=> T2 <=> T3 <=> last_node